Userland/Libraries/LibWeb/Painting/PaintableFragment.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2024, Aliaksandr Kalenik <kalenik.aliaksandr@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibWeb/Layout/LineBoxFragment.h>
+#include <LibWeb/Layout/Node.h>
+#include <LibWeb/Painting/BorderRadiiData.h>
+#include <LibWeb/PixelUnits.h>
+
+namespace Web::Painting {
+
+class PaintableFragment {
+public:
+    explicit PaintableFragment(Layout::LineBoxFragment const&);
+
+    Layout::Node const& layout_node() const { return m_layout_node; }
+
+    int start() const { return m_start; }
+    int length() const { return m_length; }
+
+    CSSPixels baseline() const { return m_baseline; }
+    CSSPixelPoint offset() const { return m_offset; }
+    void set_offset(CSSPixelPoint offset) { m_offset = offset; }
+    CSSPixelSize size() const { return m_size; }
+
+    BorderRadiiData const& border_radii_data() const { return m_border_radii_data; }
+    void set_border_radii_data(BorderRadiiData const& border_radii_data) { m_border_radii_data = border_radii_data; }
+
+    CSSPixelRect const absolute_rect() const;
+
+    Vector<Gfx::DrawGlyphOrEmoji> const& glyph_run() const { return m_glyph_run; }
+
+    CSSPixelRect selection_rect(Gfx::Font const&) const;
+
+    bool contained_by_inline_node() const { return m_contained_by_inline_node; }
+    void set_contained_by_inline_node() { m_contained_by_inline_node = true; }
+
+    CSSPixels width() const { return m_size.width(); }
+    CSSPixels height() const { return m_size.height(); }
+
+    int text_index_at(CSSPixels) const;
+
+private:
+    JS::NonnullGCPtr<Layout::Node const> m_layout_node;
```

Comment:
What protects this from GC?
Can we get rid of this pointer somehow?

Reply:
> What protects this from GC?

right, nothing has been protecting it. now it is visited from PaintableWithLines.

> Can we get rid of this pointer somehow?

we would have to do a lot more work to get rid of this pointer. currently it is used in many places during painting and hit testing.